buildengine/engine.go

```diff
@@ -32,6 +32,11 @@ type schemaChange struct {
 	*schema.Module
 }
 
+type projectMeta struct {
+	project            Project
+	lastBuildStartTime time.Time
+}
```

Comment:
@alecthomas I moved away from embedding because it seems to cause more confusion than help. For example some of the methods that took in a `project Project` would be ok with a `projectMeta` but then the type assertions would require a `projectMeta` case and I was getting lots of bugs where I should be using `project.project` but I was just using `project` directly. lemme know if this seems reasonable :)